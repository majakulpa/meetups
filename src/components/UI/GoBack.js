import React from 'react'
import { useHistory } from 'react-router-dom'
import { HiArrowLeft } from 'react-icons/hi'

const goBack = () => {
  let history = useHistory()

  return (
    <div className="text-gray-600 pt-3 pl-3">
      <button
        onClick={() => history.goBack()}
        className="flex hover:text-gray-800 hover:underline focus:outline-none"
      >
        <HiArrowLeft className="mt-1 mr-1" /> <span>Go back</span>
      </button>
    </div>
  )
}

export default goBack
