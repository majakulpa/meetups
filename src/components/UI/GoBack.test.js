import React from 'react'
import ReactDOM from 'react-dom'
import '@testing-library/jest-dom/extend-expect'
import { render, cleanup } from '@testing-library/react'
import GoBack from './GoBack'
import renderer from 'react-test-renderer'

afterEach(cleanup)

test('if renders', () => {
  const div = document.createElement('div')
  ReactDOM.render(<GoBack />, div)
})

test('if renders link correctly', () => {
  const { getByText } = render(<GoBack />)
  const linkElement = getByText(/Go back/i)
  expect(linkElement).toBeInTheDocument()
})

test('if matches snapshot', () => {
  const tree = renderer.create(<GoBack />).toJSON()
  expect(tree).toMatchSnapshot()
})
