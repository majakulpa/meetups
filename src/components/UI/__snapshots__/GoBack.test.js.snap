// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`if matches snapshot 1`] = `
<div
  className="text-gray-600 pt-3 pl-3"
>
  <button
    className="flex hover:text-gray-800 hover:underline focus:outline-none"
    onClick={[Function]}
  >
    <svg
      className="mt-1 mr-1"
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 20 20"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        clipRule="evenodd"
        d="M9.707 16.707a1 1 0 01-1.414 0l-6-6a1 1 0 010-1.414l6-6a1 1 0 011.414 1.414L5.414 9H17a1 1 0 110 2H5.414l4.293 4.293a1 1 0 010 1.414z"
        fillRule="evenodd"
      />
    </svg>
     
    <span>
      Go back
    </span>
  </button>
</div>
`;
